docs/en/index.md

```diff
@@ -159,39 +159,265 @@ Nested GridFields
+use SilverStripe\Forms\GridField\GridFieldConfig_RecordEditor;
+use SilverStripe\ORM\DataObject;
+use Symbiote\GridFieldExtensions\GridFieldNestedForm;
+
+class ChildNode extends DataObject
+{
+    //...
+
+    private static $has_one = [
+        'ParentNode' => ParentNode::class,
+    ];
+
+    private static $has_many = [
+        'GrandChildNodes' => GrandChildNode::class,
+    ];
+
+    public function getNestedConfig(): GridFieldConfig
+    {
+        $config = new GridFieldConfig_RecordEditor();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('GrandChildNodes'));
+
+        return $config;
+    }
 }
 ```
+
+```php
+namespace App\Models;
+
+use SilverStripe\ORM\DataObject;
+
+class GrandChildNode extends DataObject
+{
+    //...
+
+    private static $has_one = [
+        'ChildNode' => ChildNode::class,
+    ];
+}
+```
+
+```php
+namespace App\Admin;
+
+use App\Models\ParentNode;
+use SilverStripe\Admin\ModelAdmin;
+use SilverStripe\Forms\GridField\GridFieldConfig;
+use Symbiote\GridFieldExtensions\GridFieldNestedForm;
+
+
+class MyAdminSection extends ModelAdmin
+{
+    private static string $url_segment = 'my-admin-section';
+    private static string $menu_title = 'My Admin Section';
+
+    private static array $managed_models = [
+        ParentNode::class
+    ];
+
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('ChildNodes'));
+
+        return $config;
+    }
+}
+```
+
+There is also the possibility to use Nested GridField with the data structure ArrayList. To do this, you can use the following approach.
+
+```php
+namespace App\Models;
+
+use MyDataObject;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataObject;
+
+class MyDataSet extends DataObject
+{
+    //...
+
+    public function getMyArrayList() {
+        $list = ArrayList::create();
+        $data = MyDataObject::get();
+
+        foreach ($data as $value) {
+            $list->push($value);
+        }
+
+        return $list;
+    }
+}
+```
+
+```php
+namespace App\Admin;
+
+use App\Models\MyDataSet;
+use SilverStripe\Admin\ModelAdmin;
+use SilverStripe\Forms\GridField\GridFieldConfig;
+use Symbiote\GridFieldExtensions\GridFieldNestedForm;
+
+class MyAdminSection extends ModelAdmin
+{
+    //...
+
+    private static array $managed_models = [
+        MyDataSet::class
+    ];
+
+    protected function getGridFieldConfig(): GridFieldConfig
+    {
+        $config = parent::getGridFieldConfig();
+        $config->addComponent(GridFieldNestedForm::create()->setRelationName('getMyArrayList'));
+
+        return $config;
+    }
+}
+```
+
+Notice that instead of the name of a relation, we're passing a method name into `getMyArrayList()`. That method must return an instance of [`SS_List`](api:SilverStripe\ORM\SS_List) to be used in the GridField.
```

Comment:
```suggestion
Notice that instead of the name of a relation, we're passing a method name into `getMyArrayList()`. That method must return an instance of `SS_List` to be used in the GridField.
```

Reply:
Updated